CMakeLists.txt

```diff
@@ -109,6 +109,15 @@ add_library(zephyr_interface INTERFACE)
 # flags that come with zephyr_interface.
 zephyr_library_named(zephyr)
 
+if(CONFIG_LEGACY_GENERATED_INCLUDE_PATH)
+  zephyr_include_directories(${PROJECT_BINARY_DIR}/include/generated/zephyr)
+  message(WARNING "
+    Warning: CONFIG_LEGACY_GENERATED_INCLUDE_PATH is currently enabled by default
+    so that user applications can continue to use the legacy include paths for the
+    generated headers. This Kconfig will be deprecated and eventually removed in
+    the future releases.")
```

Comment:
Wait... there are _other_ people who look at build warnings? @henrikbrixandersen, you and I should start a club!

I've seen Kconfig and Device tree warnings pretty much every time I build anything more complex than samples/hello_world/... neither of these tools seems to have any `-Werror`, which in my experience is the only way to make anyone care.

Reply:
The warning will be printed only for people who should fix their code ASAP so printing it every time looks good to me.

Reply:
Otherwise the `CONFIG_LEGACY_GENERATED_INCLUDE_PATH` which is enabled by default now will just make everything passes silently. Most people might not notice that the headers are now relocated to `zephyr/` unless they looked at their build directory or the migration guide.

On the other extreme, if we disable `CONFIG_LEGACY_GENERATED_INCLUDE_PATH` by default, the downstream apps will fail to build immediately, which I don't think is the right way to do in an OS of this scale.

The current implementation IMHO is the middle approach - we don't want to break downstream applications, but at the same time we want to make sure that developers are made aware of the changes, and given enough time to do their migration (as we do ours - some of the PRs aren't merged, `west.yml` not updated to point at them).